src/CMakeLists.txt

```diff
@@ -1,60 +1,85 @@
+# find required packages for base and app
+find_package(LibtorrentRasterbar QUIET ${requiredLibtorrentVersion} COMPONENTS torrent-rasterbar)
+if(UNIX)
+    if(NOT LibtorrentRasterbar_FOUND)
+        include(FindPkgConfig)
+        pkg_check_modules(LIBTORRENT_RASTERBAR REQUIRED IMPORTED_TARGET GLOBAL "libtorrent-rasterbar>=${requiredLibtorrentVersion}")
+        add_library(LibtorrentRasterbar::torrent-rasterbar ALIAS PkgConfig::LIBTORRENT_RASTERBAR)
+        # force a fake package to show up in the feature summary
+        set_property(GLOBAL APPEND PROPERTY
+            PACKAGES_FOUND
+            "LibtorrentRasterbar via pkg-config (required version >= ${requiredLibtorrentVersion})"
+        )
+        set_package_properties("LibtorrentRasterbar via pkg-config (required version >= ${requiredLibtorrentVersion})"
+            PROPERTIES
+            TYPE REQUIRED
+        )
+    else()
+        set_package_properties(LibtorrentRasterbar PROPERTIES TYPE REQUIRED)
+    endif()
+else()
+    find_package(LibtorrentRasterbar ${requiredLibtorrentVersion} REQUIRED COMPONENTS torrent-rasterbar)
+endif()
+find_package(Boost ${requiredBoostVersion} REQUIRED COMPONENTS system)
```

Comment:
Should you move "find package" for all common packages into qbt_base?

Reply:
@glassez do you mean into the CML file at `src/base`, or just the top-level CML file?

I did it this way initially because:
- Both `src/base` and `src/app` need some of these packages
- Seemed awkward to have `find_package()` calls in the top-level CML, since everywhere else the calls are only made as-needed, which I think is a good thing. For example, having the `find_package()` calls to the GUI stuff in the `src/gui` CML simplifies the logic in the parent CMLs (no need for `if(GUI) find_package() ...`).

Reply:
> do you mean into the CML file at src/base,

Yes. Shouldn't `qbt_base` care about required packages? Then it can provide all common libraries/definitions as part of its interface (IIRC, it already does this job except finding packages).
>Both `src/base` and `src/app` need some of these packages

But `qbt_app` depends on `qbt_base` so it shouldn't be a problem to obtain all needed stuff via `qbt_base` interface.
(I think you understand what "interface" means in context of CMake imported targets)

Reply:
On the other hand, we can have all these "find_package" calls in one place (and GUI-related too, to be consistent). In addition, splitting the project into multiple targets (static libraries) still doesn't look certain to me.

>For example, having the `find_package()` calls to the GUI stuff in the `src/gui` CML simplifies the logic in the parent CMLs (no need for `if(GUI) find_package() ...`).

But you still have `if(GUI)` to include GUI stuff, isn't it? So it can just be written as:
```cmake
if (GUI) 
    find_package(SOME_GUI_PACKAGE)
    include(gui)
endif()
```

Reply:
@glassez 

> Yes. Shouldn't `qbt_base` care about required packages? Then it can provide all common libraries/definitions as part of its interface (IIRC, it already does this job except finding packages).

> But `qbt_app` depends on `qbt_base` so it shouldn't be a problem to obtain all needed stuff via `qbt_base` interface.
> (I think you understand what "interface" means in context of CMake imported targets)

> In addition, splitting the project into multiple targets (static libraries) still doesn't look certain to me.

Do you propose every target (gui, webui, ...) should be an "Interface Library"? If I understand correctly, using static libraries gives us more control with the transitivity of dependencies (can use PRIVATE, PUBLIC, INTERFACE). Furthermore, just for fun I tried making "qbt_base" an Interface Library, but the configure generates 3x repeated compile commands for each file of it - one per target that depends on it (qbt_gui, qbt_app and qbt_webui). So, for example, `src/base/bittorrent/resumedatasavingmanager.cpp` would be compiled 3x. This makes sense because Interface libraries don't generate build output, so each dependent must generate the build outputs themselves.

> On the other hand, we can have all these "find_package" calls in one place (and GUI-related too, to be consistent).

> But you still have `if(GUI)` to include GUI stuff, isn't it? So it can just be written as:
> 
> ```cmake
> if (GUI) 
>     find_package(SOME_GUI_PACKAGE)
>     include(gui)
> endif()
> ```

I suppose this is reasonable :+1:

Reply:
>Do you propose every target (gui, webui, ...) should be an "Interface Library" 

No.

Reply:
>I suppose this is reasonable 👍

Then all find_package calls should be moved in this script (src/CMakeLists.txt) in the same way.

Reply:
@glassez with the latest fixups, all package used under the `src/` tree are found with `find_package` calls in `src`'s CML. There is only one remaining `find_package` call, in the CML in `dist/unix/`. But `dist/` is not a subdir of `src` (they're at the same hierarchical level), so I think it's ok for that `find_package` call to remain where it is, or at most go up one level to `dist/`.